src/components/course/EnrollModal.jsx

```diff
@@ -46,40 +64,22 @@ const EnrollModal = ({
 }) => {
   const [isLoading, setIsLoading] = useState(false);
 
-  const getModalTexts = () => {
-    const { HAS_NO_SUBSIDY, HAS_COUPON_CODE, HAS_ENTERPRISE_OFFER } = MODAL_TEXTS;
-
-    if (userSubsidyApplicableToCourse?.subsidyType === COUPON_CODE_SUBSIDY_TYPE) {
-      return {
-        buttonText: HAS_COUPON_CODE.button,
-        enrollText: HAS_COUPON_CODE.body(couponCodesCount),
-        titleText: HAS_COUPON_CODE.title,
-      };
-    }
-
-    if (userSubsidyApplicableToCourse?.subsidyType === ENTERPRISE_OFFER_SUBSIDY_TYPE) {
-      return {
-        buttonText: HAS_ENTERPRISE_OFFER.button,
-        enrollText: HAS_ENTERPRISE_OFFER.body(userSubsidyApplicableToCourse, courseRunPrice),
-        titleText: HAS_ENTERPRISE_OFFER.title,
-      };
-    }
-
-    return {
-      buttonText: HAS_NO_SUBSIDY.button,
-      enrollText: HAS_NO_SUBSIDY.body,
-      titleText: HAS_NO_SUBSIDY.title,
-    };
-  };
-
   const handleEnroll = (e) => {
     setIsLoading(true);
     if (onEnroll) {
       onEnroll(e);
     }
   };
 
-  const { titleText, enrollText, buttonText } = getModalTexts();
+  const {
+    paymentRequiredForCourse, titleText, enrollText, buttonText,
+  } = getModalTexts({ userSubsidyApplicableToCourse, couponCodesCount, courseRunPrice });
+
+  // Check whether the modal should be rendered (i.e., do not show modal if user has no applicable subsidy)
+  // as payment would be required for the learner to enroll in the course.
+  if (paymentRequiredForCourse) {
+    return null;
```

Comment:
In theory, this code path should never happen; the `EnrollModal` is not rendered unless there is a subsidy.